scheduler/src/main/java/org/pentaho/platform/scheduler2/quartz/BlockingQuartzJob.java

```diff
@@ -41,9 +42,17 @@
  */
 public class BlockingQuartzJob implements Job {
   public void execute( final JobExecutionContext jobExecutionContext ) throws JobExecutionException {
+    JobDataMap jobDataMap = null;
+    if ( jobExecutionContext.getJobDetail() != null && jobExecutionContext.getJobDetail().getJobDataMap() != null ) {
+      jobDataMap = jobExecutionContext.getJobDetail().getJobDataMap();
+    }
+    if ( jobDataMap != null ) {
+      MDCUtil.setupSchedulerMDC( jobDataMap.get( QuartzScheduler.RESERVEDMAPKEY_ACTIONUSER ), jobDataMap.get(
+          "lineage-id" ) );
+    }
     final boolean jobRestarted;
-    if ( jobExecutionContext != null && jobExecutionContext.getJobDetail() != null ) {
-      jobRestarted = jobExecutionContext.getJobDetail().getJobDataMap().getBooleanValue( QuartzScheduler.RESERVEDMAPKEY_RESTART_FLAG );
+    if ( jobDataMap != null ) {
+      jobRestarted = jobDataMap.getBooleanValue( QuartzScheduler.RESERVEDMAPKEY_RESTART_FLAG );
```

Comment:
Should we add these two new MDC properties to the log4j.xml configuration?

Reply:
They are already there in log4j.xml.  QuartzScheduler.RESERVEDMAPKEY_ACTIONUSER corresponds to sessionName and "lineage-id" corresponds to instanceId